logstash-core/lib/logstash/pipeline.rb

```diff
@@ -82,14 +82,13 @@ def initialize(config_str, settings = LogStash::SETTINGS, namespaced_metric = ni
       raise
     end
 
-    @input_queue = LogStash::Util::WrappedSynchronousQueue.new
+    queue = LogStash::Util::WrappedSynchronousQueue.new
+    @input_queue_client = queue.write_client
+    @filter_queue_client = queue.read_client
```

Comment:
+1 on these as separate objects. Eventually it'd be nice to have a separate write_client per-input. Is this where you're going with this?